src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs

```diff
@@ -1,462 +1,407 @@
         }
-    }
-#endif
 
-    [Fact]
-    [PlatformSpecific(PlatformID.Windows)] // max directory length not fixed on Unix
-    public static void CreateDirectory_DirectoryEqualToMaxDirectory_CanBeCreated()
-    {   // Recursively creates directories right up to the maximum directory length ("247 chars not including null")
-        using (TemporaryDirectory directory = new TemporaryDirectory())
+        [Fact]
+        [PlatformSpecific(PlatformID.Windows)] // testing drive labels
+        [ActiveIssue(1221)]
+        public void SubdirectoryOnNotReadyDriveAsPath_ThrowsIOException()
         {
-            PathInfo path = IOServices.GetPath(directory.Path, IOInputs.MaxDirectory, IOInputs.MaxComponent);
-
-            // First create 'C:\AAA...AA', followed by 'C:\AAA...AAA\AAA...AAA', etc
-            foreach (string subpath in path.SubPaths)
+            var drive = IOServices.GetNotReadyDrive();
+            if (drive == null)
             {
-                DirectoryInfo result = Directory.CreateDirectory(subpath);
-
-                Assert.Equal(subpath, result.FullName);
-                Assert.True(Directory.Exists(result.FullName));
+                Console.WriteLine("Skipping test. Unable to find a not-ready drive, such as CD-Rom with no disc inserted.");
+                return;
             }
+
+            // 'Device is not ready'
+            Assert.Throws<IOException>(() =>
+            {
+                Create(Path.Combine(drive, "Subdirectory"));
+            });
         }
-    }
 
-    [Fact]
-    [PlatformSpecific(PlatformID.Windows)] // max directory length not fixed on Unix
-    public static void CreateDirectory_DirectoryEqualToMaxDirectory_CanBeCreatedAllAtOnce()
-    {   // Creates directories up to the maximum directory length all at once
-        using (TemporaryDirectory directory = new TemporaryDirectory())
+        [Fact]
+        [ActiveIssue(1221)]
+        [PlatformSpecific(PlatformID.Windows)] // testing drive labels
+        public void NonExistentDriveAsPath_ThrowsDirectoryNotFoundException()
         {
-            PathInfo path = IOServices.GetPath(directory.Path, IOInputs.MaxDirectory, maxComponent: 10);
-
-            DirectoryInfo result = Directory.CreateDirectory(path.FullPath);
+            var drive = IOServices.GetNonExistentDrive();
+            if (drive == null)
+            {
+                Console.WriteLine("Skipping test. Unable to find a non-existent drive.");
+                return;
+            }
 
-            Assert.Equal(path.FullPath, result.FullName);
-            Assert.True(Directory.Exists(result.FullName));
+            Assert.Throws<DirectoryNotFoundException>(() =>
+            {
+                Create(drive);
+            });
         }
-    }
 
-    [Fact]
-    public static void CreateDirectory_ValidPathWithTrailingSlash_CanBeCreated()
-    {
-        using (TemporaryDirectory directory = new TemporaryDirectory())
+        [Fact]
+        [ActiveIssue(1221)]
```

Comment:
Yeah, I struggled deciding on this one. Since I have no real idea of what the final mounted drive testing framework is going to look like, I removed these tests to be safe. Realistically, they're fine in the form:
```
Assert.Throws<DirectoryNotFoundException>(() =>
{
       Create(IOServices.GetNonExistentDrive());
});
```
They will only fail if IOServices finds no NonExistentDrives, in which case the error would be ArgumentNullException instead of DirectoryNotFoundException. Although if that is the case, then there's something wrong with the testing system that is disallowing proper drive testing <i>anyways</i>, so it should be fine to reinclude them.